lib/gamnit/depth/more_models.nit

```diff
@@ -0,0 +1,386 @@
+
+					vertices.add v.x
+					vertices.add v.y
+					vertices.add v.z
+				end
+			end
+
+			# TODO use polygon triangulation to support larger polygons
+		end
+		return vertices
+	end
+
+	# Compute the normals of `faces` in a flat `Array[Float]`
+	fun normals(faces: Array[ObjFace]): Array[Float] do
+		var obj_def = obj_def
+
+		var normals = new Array[Float]
+		for face in faces do
+			# 1st triangle
+			var count = 0
+			for e in face.vertices do
+				var i = e.normal_index
+				if i == null then
+					compute_and_append_normal(normals, face)
+				else
+					var v = obj_def.normals[i-1]
+					normals.add v.x
+					normals.add v.y
+					normals.add v.z
+				end
+
+				if count == 2 then break
+				count += 1
+			end
+
+			# If square, 2nd triangle
+			#
+			# This may not support all vertices ordering.
+			if face.vertices.length > 3 then
+				for e in [face.vertices[0], face.vertices[2], face.vertices[3]] do
+					var i = e.normal_index
+					if i == null then
+						compute_and_append_normal(normals, face)
+					else
+						var v = obj_def.normals[i-1]
+						normals.add v.x
+						normals.add v.y
+						normals.add v.z
+					end
+				end
+			end
+		end
+		return normals
+	end
+
+	# Compute the normal of `face` and append it as 3 floats to `seq`
+	#
+	# Resulting normals are not normalized.
+	fun compute_and_append_normal(seq: Sequence[Float], face: ObjFace)
+	do
+		var i1 = face.vertices[0].vertex_point_index
+		var i2 = face.vertices[1].vertex_point_index
+		var i3 = face.vertices[2].vertex_point_index
+
+		var v1 = obj_def.vertex_points[i1-1]
+		var v2 = obj_def.vertex_points[i2-1]
+		var v3 = obj_def.vertex_points[i3-1]
+
+		var vx = v2.x - v1.x
+		var vy = v2.y - v1.y
+		var vz = v2.z - v1.z
+		var wx = v3.x - v1.x
+		var wy = v3.y - v1.y
+		var wz = v3.z - v1.z
+
+		var nx = (vy*wz) - (vz*wy)
+		var ny = (vz*wx) - (vx*wz)
+		var nz = (vx*wy) - (vy*wx)
+
+		# Append to `seq`
+		seq.add nx
+		seq.add ny
+		seq.add nz
+	end
+
+	# Compute the texture coordinates of `faces` in a flat `Array[Float]`
+	fun texture_coords(faces: Array[ObjFace]): Array[Float] do
+		var obj_def = obj_def
+
+		var coords = new Array[Float]
+		for face in faces do
+
+			# 1st triangle
+			var count = 0
+			for e in face.vertices do
+				var i = e.texture_coord_index
+				if i == null then
+					coords.add 0.0
+					coords.add 0.0
+				else
+					var tc = obj_def.texture_coords[i-1]
+					coords.add tc.u
+					coords.add tc.v
+				end
+
+				if count == 2 then break
+				count += 1
+			end
+
+			# If square, 2nd triangle
+			#
+			# This may not support all vertices ordering.
+			if face.vertices.length > 3 then
+				for e in [face.vertices[0], face.vertices[2], face.vertices[3]] do
+					var i = e.texture_coord_index
+					if i == null then
+						coords.add 0.0
+						coords.add 0.0
+					else
+					var tc = obj_def.texture_coords[i-1]
+					coords.add tc.u
+					coords.add tc.v
+					end
+				end
+			end
+		end
+		return coords
+	end
+end
+
+redef class Sys
```

Comment:
As @BlackMinou demonstrated with `Sound`, App is not a good container for automatic lists of assets.

Most assets are attributes of `App`, so if the asset's init accesses `Sys::app` it causes a recursive infinite loop.